cypress/e2e/lists.cy.ts

```diff
@@ -0,0 +1,101 @@
+    it('should allow to filter by date', () => {
+        cy.visit('/');
+        login();
+        cy.findByText('Contacts').click();
+        getPaginationText().then(el => {
+            const count = parseInt(el.text().split('of')[1].trim());
+
+            cy.findByText('Before last month').click();
+            // Use should here to allow built-in retry as it may take a few ms for the list to update
+            getPaginationText().should(el => {
+                const countFiltered = parseInt(el.text().split('of')[1].trim());
+                expect(countFiltered).to.be.lessThan(count);
+            });
+        });
+    });
+
+    it('should allow to filter by status', () => {
+        cy.visit('/');
+        login();
+        cy.findByText('Contacts').click();
+        getPaginationText().then(el => {
+            const count = parseInt(el.text().split('of')[1].trim());
+
+            cy.findByText('Cold').click();
+            // Use should here to allow built-in retry as it may take a few ms for the list to update
+            getPaginationText().should(el => {
+                const countFiltered = parseInt(el.text().split('of')[1].trim());
+                expect(countFiltered).to.be.lessThan(count);
+            });
+        });
+    });
+
+    it('should allow to filter by tag', () => {
+        cy.visit('/');
+        login();
+        cy.findByText('Contacts').click();
+        getPaginationText().then(el => {
+            const count = parseInt(el.text().split('of')[1].trim());
+
+            cy.findByText('football-fan', {
+                selector: '[role=button] *',
+            }).click();
+
+            // Use should here to allow built-in retry as it may take a few ms for the list to update
+            getPaginationText().should(el => {
+                const countFiltered = parseInt(el.text().split('of')[1].trim());
+                expect(countFiltered).to.be.lessThan(count);
+            });
+        });
+    });
+
+    it('should allow to move through pages', () => {
+        cy.visit('/');
+        login();
+        cy.findByText('Contacts').click();
+        getPaginationText().then(el => {
+            const page = parseInt(el.text().split('-')[0].trim());
+            cy.findByLabelText('Go to page 4').click();
+
+            let page6 = 0;
+            // Use should here to allow built-in retry as it may take a few ms for the list to update
+            getPaginationText().should(el => {
+                page6 = parseInt(el.text().split('-')[0].trim());
+                expect(page6).to.be.greaterThan(page);
+            });
+
+            cy.findByLabelText('Go to page 2').click();
+
+            // Use should here to allow built-in retry as it may take a few ms for the list to update
+            getPaginationText().should(el => {
+                const page2 = parseInt(el.text().split('-')[0].trim());
+                expect(page2).to.be.greaterThan(page);
+                expect(page2).to.be.lessThan(page6);
```

Comment:
Nitpicking but it's actually `page4` :sweat_smile: 

```suggestion
            let page4 = 0;
            // Use should here to allow built-in retry as it may take a few ms for the list to update
            getPaginationText().should(el => {
                page4 = parseInt(el.text().split('-')[0].trim());
                expect(page4).to.be.greaterThan(page);
            });

            cy.findByLabelText('Go to page 2').click();

            // Use should here to allow built-in retry as it may take a few ms for the list to update
            getPaginationText().should(el => {
                const page2 = parseInt(el.text().split('-')[0].trim());
                expect(page2).to.be.greaterThan(page);
                expect(page2).to.be.lessThan(page4);
```

Reply:
Besides, I realize that you could just expect `page2` to equal `2` and `page4` to equal `4`, can't you?

Reply:
Nope that's more something like `25-50 of 150`

Reply:
Oh right, my bad